open-craft/configuration
# FAL-2260 Prepare opencraft release branch for lilac.2

[FAL-2260](https://tasks.opencraft.com/browse/FAL-2260)

Port the following PRs / commits from koa to lilac:

- https://github.com/open-craft/configuration/pull/162
- https://github.com/open-craft/configuration/pull/164
- https://github.com/open-craft/configuration/pull/159
- https://github.com/open-craft/configuration/pull/167
- https://github.com/open-craft/configuration/pull/169
- https://github.com/edx/configuration/commit/4df1323d8906675147acb7c448246808e334e3fd
- https://github.com/open-craft/configuration/pull/173

These are skipped because they have already been upstreamed in lilac :tada: :

- https://github.com/open-craft/configuration/pull/170

And this isn't required any more (see comments below):

- https://github.com/open-craft/configuration/pull/140

**Test instructions**:

For each patch from https://github.com/edx/configuration/compare/open-release/koa.master...open-craft:opencraft-release/koa.3 , verify that it has either been upstreamed or ported cleanly.

Launch a lilac instance with ocim, using this configuration branch, and verify everything is deployed successfully. (wip - deploying on https://manage.opencraft.com/instance/29580/ )

**Notes / Concerns**:

- [x] Not sure about https://github.com/open-craft/configuration/pull/175/commits/93ba25ec09f6f071e54a984c1f26721dc0bf1d2c - we don't need to use our fork of tubular any more, since upstream has merged https://github.com/edx/tubular/pull/532 .  The upstream fix is in lilac.master though, not lilac.2.  I'd recommend modifying that commit to pin to edx/tubular@open-release/lilac.master , but not sure if that's what we want to do here.  I'm not sure why we're pinning it here, and not in private extra-configuration. CC @lgp171188 
- [x] Also not sure about https://github.com/open-craft/configuration/pull/140 - upstream had changed the import paths CC @kaizoku
- [x] I couldn't find an associated PR for https://github.com/edx/configuration/commit/4df1323d8906675147acb7c448246808e334e3fd CC @pkulkark could you check this?
